src/lib/components/SideNav/index.js

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {html} from "lit-element";
+import {BaseElement} from "../BaseElement";
+import {store} from "../../store";
+import "wicg-inert";
+import {closeSideNav} from "../../actions";
+
+class SideNav extends BaseElement {
+  static get properties() {
+    return {
+      animatable: {type: Boolean, reflect: true},
+      expanded: {type: Boolean, reflect: true},
+    };
+  }
+
+  constructor() {
+    super();
+
+    this.inert = true;
+    this.animatable = false;
+    this.expanded_ = false;
+    this.startX_ = 0;
+    this.currentX_ = 0;
+    this.touchingSideNav_ = false;
+    this.prerenderedChildren_ = null;
+
+    this.onCloseSideNav = this.onCloseSideNav.bind(this);
+    this.onBlockClicks = this.onBlockClicks.bind(this);
+    this.onTouchStart = this.onTouchStart.bind(this);
+    this.onTouchMove = this.onTouchMove.bind(this);
+    this.onTouchEnd = this.onTouchEnd.bind(this);
+    this.onTransitionEnd = this.onTransitionEnd.bind(this);
+    this.drag = this.drag.bind(this);
+    this.onStateChanged = this.onStateChanged.bind(this);
+  }
+
+  render() {
+    if (!this.prerenderedChildren_) {
+      this.prerenderedChildren_ = [];
+      for (const child of this.children) {
+        this.prerenderedChildren_.push(child);
+      }
+    }
+    return html`
+      <nav class="web-side-nav__container">
+        <div class="web-side-nav__header">
+          <button
+            @click=${this.onCloseSideNav}
+            class="web-side-nav__hide"
+          ></button>
+          <a
+            href="/"
+            class="gc-analytics-event"
+            data-category="Site-Wide Custom Events"
+            data-label="Site logo"
+          >
+            <img
+              class="web-side-nav__logo"
+              src="/images/lockup.svg"
+              alt="web.dev"
+            />
+          </a>
+        </div>
+        ${this.prerenderedChildren_}
+      </nav>
+    `;
+  }
+
+  firstUpdated() {
+    this.sideNavContainerEl = this.querySelector(".web-side-nav__container");
+    this.addEventListeners();
+    store.subscribe(this.onStateChanged);
+    this.onStateChanged();
+  }
+
+  addEventListeners() {
+    this.addEventListener("click", this.onCloseSideNav);
```

Comment:
For everything not on `this`, same as my other comment—shouldn't this be done inside lit?

We could even add a top-level element that's the same size that handles the other on... handlers.

Reply:
I fixed this for the handler referencing sideNavContainerEl. But I kept the addListeners block for the stuff pointing at `this`. I explored creating a wrapper element, but [lit's syntax for passing options to a declarative handler](https://github.com/Polymer/lit-html/pull/531) seems to make the code more confusing than it's worth.

Basically I was ending up creating objects that contained the touch code, and the functions within those objects have to be bound back to `this` so it's just kinda gross.